Add displayArrayKey Option for Rendering Array Keys in JSONViewer

Cylop opened the pull request:
Greetings,

I have implemented a new feature in the JSONViewer, introducing an option named displayArrayKey. This feature aims to enhance the user experience by providing additional flexibility in visualizing JSON data.

# Feature: 'displayArrayKey' Option
The displayArrayKey option, when set to false, prevents the rendering of the keys of an array and appends a colon to the end of the line. This results in a cleaner and more concise output for users who find array keys unnecessary in certain contexts.

## Usage:
```tsx
<JsonView
  src={object}
  displayArrayKey={false}
/>
```

### Enabled (without keys): 
<img width="603" alt="Bildschirmfoto 2023-09-27 um 23 39 28" src="https://github.com/YYsuni/react18-json-view/assets/19712231/ae8d63e9-d19f-49f4-a2e3-57bcbf7b5b88">

### Disabled (with keys): 
<img width="603" alt="Bildschirmfoto 2023-09-27 um 23 39 47" src="https://github.com/YYsuni/react18-json-view/assets/19712231/fdddbed4-398b-4baa-9f6d-6cece733c7a8">

## Use Case:
This addition is beneficial for developers seeking a streamlined representation of JSON data, especially when handling large arrays where the keys might not contribute significantly to comprehending the data structure.

## Documentation:
The documentation has been updated to encompass the new displayArrayKey option, elucidating its function and effect on the JSON data representation.

## Conclusion:
I am confident that this feature will serve as a valuable enhancement to the project by offering more versatility in displaying JSON data. I look forward to your feedback and am open to making any necessary modifications.

Thank you for considering this contribution. Looking forward to a positive review and eventual merge.

Best Regards,
Nicholas

YYsuni commented on package-lock.json:
Project use pnpm as default for the workspace

YYsuni replied:
package-lock file will confuse Vercel deployment.

YYsuni commented on src/components/json-node.tsx:
We just need to use `displayArrayKey` in ObjectNode component. Changes here are needless.

YYsuni commented on src/components/name-key.tsx:
I'm not sure why the new component was created here. We could just change a little code in the NameValue component.

Cylop replied:
I’ve separated the code into a distinct component for clarity and readability. Integrating it with the original component could lead to clutter. However, I can relocate this new key component to the name-value file for a more concise separation of code. Would that arrangement be more suitable for you?

YYsuni commented on src/components/name-value.tsx:
The important point is to use the displayArrayKey from useContext instead of props. Also, the conditional operator is enough.

YYsuni commented on src/components/object-node.tsx:
Again, attribute passing should not happen except for depth and props with a parent relationship.

YYsuni commented:
I noticed the new comma added and had a thought: why do we add the prop viewMode: viewer or json? Which means json mode has the comma symbol, but the viewer mode not.

Cylop commented:
That sounds like a good idea, I'll implement the comments!

Cylop commented:
I started changing right now, do you think with the view mode 'viewer' or 'json' - do we need the displayArrayKey option at all?

Cylop commented:
I introduced now the new option viewMode with options['json', 'viewer'] where 'viewer' is the default one. Once  'json' is selected, commas are appended to the end of the line, whereas 'displayArrayKey' only the display of array keys disabled. 

I removed the package-lock and used pnpm (missed that before) 

I also removed the new file and adapted the existing 'name-value.tsx'. Please provide me some feedback 😄

YYsuni commented:
Sorry, I am on vacation.

YYsuni commented on README.md:
The viewMode property seems a bit wasteful as it is only used to determine whether to display commas.😆
Let me think twice...🤔

YYsuni commented on src/components/json-view.tsx:
The formatting here is not that friendly.😅
